server/views/pages/activities/record-attendance/not-attended-reason.njk

```diff
@@ -168,17 +173,22 @@
                         {% endif %}
                     {% endfor %}
 
+                    {% set prisonerIndex = loop.index0 %}
+
+                    <input type="hidden" name="{{ "notAttendedData[" + prisonerIndex + "][prisonerNumber]" }}" value="{{ prisoner.prisonerNumber }}" />
+                    <input type="hidden" name="{{ "notAttendedData[" + prisonerIndex + "][prisonerName]" }}" value="{{ prisoner.prisonerName | toTitleCase }}" />
```

Comment:
I tried everything I could to not need this. It is not possible to provide additional context to a nested object when validating with class-validator.

This this probably the best work around to add the prisoner name to the request object.